### PR TITLE
upgrade to bluebird 3 and minor cleanup

### PR DESCRIPTION
**Pull request by khoomeister**
Investigating a bug in our app - did some upgrading & adding a test to check for memcached 1MB object size limit

### CODE REVIEW

File: test/cache.coffee [hunk 1]
```diff
@@ -55,6 +55,19 @@ describe 'Cache', ->
     it 'supports promise style', (done) ->
       @cache.set('key2', values.key2, { expire: 1 }).nodeify done
 
+  describe 'memcached', ->
```
Comment (jkrems) [inline test/cache.coffee]: Just trying to understand - is this something that changes between bluebird 2 and 3? It looks like it works "as expected" with bluebird 2. When you provide a callback, it properly "ends" the promise, and without you end up with a "Possibly unhandled rejection".

**Comment by jkrems** [comment]
Is it really a concern for `cached` that 1MB happens to be the default setting for max values in memcached? Trying to understand what the additional test is testing. :)

**Comment by khoomeister** [comment]
No you're totally right - this was originally put up to test behavior so I could debug the problem with the app using this package.

I've since removed the test code.

File: src/cache.coffee [hunk 2]
```diff
@@ -141,13 +141,12 @@ class Cache
       passThroughData
 
     writeValue = (generatedValue) =>
-      @set(rawKey, generatedValue, opts).then(
-        (rawValue) -> rawValue?.d ? null
-        ->
```
Comment (khoomeister) [inline src/cache.coffee]: This newline to specify additional args for a function definitely doesn't make code easier to read.

**Comment by jkrems** [comment]
LGTM

**Pull request merged**